src/content/docs/dev/general-message-passing/sui/gmp-tutorial.mdx

```diff
@@ -0,0 +1,918 @@
+
+The `new_transaction()` function call expects several parameters.
+
+1. `is_final: bool`: For this you simply pass `true` telling the relayer to execute the transaction.
+1. `move_calls: vector<MoveCall>`: A vector of [MoveCalls](https://github.com/axelarnetwork/axelar-cgp-sui/blob/main/move/relayer_discovery/sources/types/transaction.move#L44)
+
+```rust
+    let transaction = transaction::new_transaction(
+        true,
+        vector[
+            transaction::new_move_call(
+                transaction::new_function(
+                    address::from_bytes(
+                        hex::decode(
+                            *ascii::as_bytes(
+                                &type_name::get_address(
+                                    &type_name::get<Singleton>(),
+                                ),
+                            ),
+                        ),
+                    ),
+                    ascii::string(b"gmp"),
+                    ascii::string(b"execute"),
+                ),
+                arguments,
+                vector[],
+            ),
+        ],
+    );
+```
+
+The `is_final` can simply be set to `true`. The `move_calls()` needs to receive the `function`, `arguments`, and `type_arguments` that will be triggered on that package for the object id that is being sent by the relayer. In your case the function is the `execute` function on the `gmp` module. The `arguments` are the `ApprovedMessage` object and the `singleton` you had put together previously, and the `type_arguments` can be left empty for now.
+
+
+The output of this will be your new `transaction` that you can pass into the Relayer Discover's [register_transaction()](https://github.com/axelarnetwork/axelar-cgp-sui/blob/main/move/relayer_discovery/sources/discovery.move#L93) function along with a reference to your package's `channel` as follows: 
+
+```rust
+discovery.register_transaction(&singleton.channel, transaction);
+```
+
+Once this is called your package will be registered with the Relayer Discovery so it can begin receiving cross-chain messages.
+
+Lastly you can define your `concat()` helper function to append the arguments used in the `register_transaction()` function.
+
+```rust
+fun concat<T: copy>(v1: vector<T>, v2: vector<T>): vector<T> {
+    let mut result = v1;
+    result.append(v2);
+    result
+}
+```
+
+#### Why is this needed?
+
+In the EVM ecosystem because you inherited from AxelarExecutable, the relayer already knows your contract’s function signature and where to call it. This is because on EVM chains contracts also hold their own storage. On Sui chains package logic is separate from objects that are holding the relevant state. As a result the destination that is sent from the source chain will be the `channel_id` as opposed to the `package_id`. In other words  The package ID just tells you which code to run; the channel ID tells you where to run it and which messages to pull in
```

Comment:
```suggestion
In the EVM ecosystem, because you inherited from `AxelarExecutable`, the relayer already knows your contract’s function signature and where to call it. This is because contracts also hold their storage on EVM chains. On Sui chains, the package logic is separate from objects that hold the relevant state. As a result, the destination sent from the source chain will be the `channel_id` instead of the `package_id`. In other words, the package ID tells you which code to run; the channel ID tells you where to run it and which messages to pull in.
```